src/OpenLoco/src/GameCommands/Terraform/LowerRaiseLandMountain.cpp

```diff
@@ -0,0 +1,405 @@
+        adjustSurfaceSlopeNorth(targetPos, height, removedBuildings);
+    }
+
+    static void smoothenSurfaceEast(const Pos2& refPos, const Pos2& targetPos, const LowerRaiseLandMountainArgs& args, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        auto tile = TileManager::get(refPos);
+        auto surface = tile.surface();
+
+        auto height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::west);
+        adjustSurfaceSlopeSouth(targetPos, height, removedBuildings);
+
+        *mtnToolHeightDiff -= kSmallZStep;
+
+        if (targetPos.x >= args.pointA.x)
+        {
+            *mtnToolHeightDiff += kSmallZStep;
+            if (targetPos.x > args.pointB.x)
+            {
+                *mtnToolHeightDiff += kSmallZStep;
+            }
+        }
+
+        height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::north);
+        adjustSurfaceSlopeEast(targetPos, height, removedBuildings);
+    }
+
+    static void smoothenSurfaceSouth(const Pos2& refPos, const Pos2& targetPos, const LowerRaiseLandMountainArgs& args, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        auto tile = TileManager::get(refPos);
+        auto surface = tile.surface();
+
+        auto height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::north);
+        adjustSurfaceSlopeWest(targetPos, height, removedBuildings);
+
+        *mtnToolHeightDiff -= kSmallZStep;
+
+        if (targetPos.y <= args.pointB.y)
+        {
+            *mtnToolHeightDiff += kSmallZStep;
+            if (targetPos.y < args.pointA.y)
+            {
+                *mtnToolHeightDiff += kSmallZStep;
+            }
+        }
+
+        height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::east);
+        adjustSurfaceSlopeSouth(targetPos, height, removedBuildings);
+    }
+
+    static void smoothenSurfaceWest(const Pos2& refPos, const Pos2& targetPos, const LowerRaiseLandMountainArgs& args, std::set<Pos3, LessThanPos3>& removedBuildings)
+    {
+        auto tile = TileManager::get(refPos);
+        auto surface = tile.surface();
+
+        auto height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::east);
+        adjustSurfaceSlopeNorth(targetPos, height, removedBuildings);
+
+        *mtnToolHeightDiff -= kSmallZStep;
+
+        if (targetPos.x <= args.pointB.x)
+        {
+            *mtnToolHeightDiff += kSmallZStep;
+            if (targetPos.x < args.pointA.x)
+            {
+                *mtnToolHeightDiff += kSmallZStep;
+            }
+        }
+
+        height = TileManager::getSurfaceCornerHeight(*surface, SurfaceSlope::CornerUp::south);
+        adjustSurfaceSlopeWest(targetPos, height, removedBuildings);
+    }
+
+    // 0x00462DCE
+    static uint32_t lowerRaiseLandMountain(const LowerRaiseLandMountainArgs& args, const uint8_t flags)
+    {
+        mtnToolXParams = (args.pointB.x << 16) | args.pointA.x;
```

Comment:
are these still used somewhere?

Reply:
Ah, no, they can be removed. I haven't refactored out the flags and cost globals yet though. I'd prefer that to be done at a later point.